src/test/java/HeroMapperTest.java

```diff
@@ -1,2 +1,184 @@
+    }
+
+    @Test
+    public void correctHeroFollowersListSizeFetchedToDataModel() {
+        Mockito.when(testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock)).thenReturn(heroDataModel);
+        Assertions.assertEquals(1, testHeroMapperMock.fetchHeroToDataModel(battleTag, heroId, testFetchTokenMock).getFollowers().size());
+    }
+
+    @Test
+    public void wrongBattleTagFormatProvided() {
+        Mockito.when(testHeroMapperMock.generateRequest(battleTagAsWrongFormat, heroId, testFetchTokenMock))
+                .thenReturn("Niepoprawny format battleTag! Spróbuj ponownie.");
+
+        Assertions.assertTrue(testHeroMapperMock.generateRequest(battleTagAsWrongFormat, heroId, testFetchTokenMock)
+                .contains(wrongBattleTagFormatWarning));
+    }
+
+    @Test
+    public void wrongHeroIdFormatProvided() {
+        Mockito.when(testHeroMapperMock.generateRequest(battleTag, heroIdAsWrongFormat, testFetchTokenMock))
+                .thenReturn("Niepoprawny format heroId - tylko cyfry! Spróbuj ponownie.");
+
+        Assertions.assertTrue(testHeroMapperMock.generateRequest(battleTag, heroIdAsWrongFormat, testFetchTokenMock)
+                .contains(wrongHeroIdFormatWarning));
+    }
+
+    @Test
+    public void providedBattleTagDoesntExistThrowsException() {
+        Exception exception = Assertions.assertThrows(RuntimeException.class, ()-> {
+            testHeroMapper.fetchHeroToDataModel(battleTagAsDosentExist, heroId, testFetchToken);
+        });
+
+        String expectedMessage = "Gracz o podanym battleTagu nie istnieje w swiecie Sanktuarium!";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void providedBattleTagIsNullThrowsException() {
+        Exception exception = Assertions.assertThrows(NullPointerException.class, ()-> {
+            testHeroMapper.fetchHeroToDataModel(battleTagAsNull, heroId, testFetchTokenMock);
+        });
+
+        String expectedMessage = "Cannot invoke \"String.replace(char, char)\" because \"battleTag\" is null";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void providedHeroIdIsNullThrowsException() {
+        Exception exception = Assertions.assertThrows(IllegalArgumentException.class, ()-> {
+            testHeroMapper.fetchHeroToDataModel(battleTag, heroIdAsNull, testFetchTokenMock);
+        });
+
+        String expectedMessage = "argument \"content\" is null";
+        String actualMessage = exception.getMessage();
+
+        Assertions.assertTrue(actualMessage.contains(expectedMessage));
+    }
+
+    @Test
+    public void providedTokenNullThrowsException() {
+        Exception exception = Assertions.assertThrows(NullPointerException.class, ()-> {
```

Comment:
assertThrows suffices for exception tests

Reply:
done